Dockerfile.dev

```diff
@@ -9,14 +9,11 @@ WORKDIR /go/src/github.com/netlify/gotrue
 
 # Pulling dependencies
 COPY ./Makefile ./go.* ./
-
-# Production dependencies
 RUN make deps
 
-# Development dependences
-RUN go get -u github.com/Thatooine/go-test-html-report
-RUN go get github.com/githubnemo/CompileDaemon
-RUN go install github.com/githubnemo/CompileDaemon
+# Building stuff
+COPY . /go/src/github.com/netlify/gotrue
+RUN make build
 
 RUN adduser -D -u 1000 netlify
```

Comment:
I'm surprised you're having permissions issues with the hot reload. I wonder if it has to do with switching to the netlify user after copying the project files in. Maybe try switching to the netlify user before copying the files or removing the netlify user since this is a development container.